src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+```
+
+An approximately 5-fold difference in the library sizes means that we will need to apply a normalization method before continuing with the analysis. The most common normalization methods used in microbiome count data are proportions and rarefaction. However, other methods originally developed to normalize RNA-seq counts have been adapted to differential-abundance analysis in microbiome data. A discussion about how to choose the right normalization method is out of the scope of this material, but the topic has been covered in multiple forums and scientific publications. Depending on the downstream analysis we intend to perform, different methods might be appropriate. For instance, to compare groups of samples at community-level through beta-diversity, “…proportions and rarefying produced more accurate comparisons among communities and are the only methods that fully normalized read depths across samples. Additionally, upper quartile, cumulative sum scaling (CSS), edgeR-TMM, and DESeq-VS often masked differences among communities when common OTUs differed, and they produced false positives when rare OTUs differed” ([2](#ref-mcknight2019)). On the other hand, for detection of differentially abundant species, “both proportions and rarefied counts result in a high rate of false positives in tests for species that are differentially abundant across sample classes” ([3](#ref-mcmurdie2014)).
+
+In the following examples we will show three popular ways of normalization: relative abundance, rarefaction and cummulative sum scaling.
+
+## Normalization by total sum scaling (TSS, relative abundance or proportions)
+The simplest way to normalize the differences in sample size is to transform the OTU counts table into relative abundance by dividing by the number of total reads of each sample. This type of normalization is also referred to as relative abundance or proportions. We use this normalization to compare taxonomic profiles, while alpha diversity indices are computed on the non-normalized matrix. The reason to do so is that we need a matrix of integer numbers as input.
+
+1. Transform taxonomy raw-counts matrix into relative abundance.
+
+```{r}
+filtered_tse <- transformAssay(filtered_tse, assay.type = "counts", name = "relabundance", method = "relabundance")
+
+```
+
+2. Agglomerate by class rank to the `glom_tse`.
+
+```{r}
+glom_tse <- agglomerateByRank(filtered_tse, rank = "Class", assay.type = "relabundance")
+
+```
+
+3. Get the top 15 features and store them in the `tss_tse`. We store the top features in a new tse object as keeping the `glom_tse` and `tss_tse` objects  will come in handy later on.
+
+```{r}
+# Take the top 15 features based on their means
+top15_tss <- getTopFeatures(glom_tse, method = "mean", assay.type = "relabundance", top = 15L)
+
+# Store those top 15 features and all relevant metadata in a new tse
+tss_tse <- TreeSummarizedExperiment(
```

Comment:
Is the idea here just to subset the original tse?

Then whynot just do 

`tse.tss <- glom_tse[top15_tss,]`

Reply:
yeah that was dumb changed that earlier today